apps/studio/components/interfaces/BranchManagement/BranchStatusBadge.tsx

```diff
@@ -0,0 +1,45 @@
+import { BranchData } from 'data/branches/branch-query'
+import { AlertTriangleIcon, ClockIcon } from 'lucide-react'
+import { Badge } from 'ui'
+
+type Status = BranchData['status']
+
+export interface BranchStatusBadgeProps {
+  status: Status
+}
+
+const UNHEALTHY_STATUES: Status[] = ['ACTIVE_UNHEALTHY', 'INIT_FAILED', 'UNKNOWN']
+const WAITING_STATUSES: Status[] = ['COMING_UP', 'GOING_DOWN', 'PAUSING', 'RESTORING', 'UPGRADING']
+
+const STATUS_TO_LABEL: Record<Status, string> = {
```

Comment:
can we use PROJECT_STATUS actually? seems like there's quite an overlap

Reply:
There will be more branching related statuses added here in future

Reply:
resolving